bezier-rs/lib/src/lib.rs

```diff
@@ -453,15 +460,89 @@ impl Bezier {
 			.try_into()
 			.unwrap()
 	}
+
+	/// Returns a Bezier curve that results from applying the tranformation function to each point in the Bezier.
+	pub fn apply_transformation(&self, transformation_function: &dyn Fn(DVec2) -> DVec2) -> Bezier {
+		let transformed_start = transformation_function(self.start);
+		let transformed_end = transformation_function(self.end);
+		match self.handles {
+			BezierHandles::Quadratic { handle } => {
+				let transformed_handle = transformation_function(handle);
+				Bezier::from_quadratic_dvec2(transformed_start, transformed_handle, transformed_end)
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				let transformed_handle_start = transformation_function(handle_start);
+				let transformed_handle_end = transformation_function(handle_end);
+				Bezier::from_cubic_dvec2(transformed_start, transformed_handle_start, transformed_handle_end, transformed_end)
+			}
+		}
+	}
+
+	/// Returns a Bezier curve that results from rotating the curve by the given angle (in radians).
+	pub fn rotate(&self, angle: f64) -> Bezier {
+		let rotation_matrix = DMat2::from_angle(angle);
+		self.apply_transformation(&|point| rotation_matrix.mul_vec2(point))
+	}
+
+	/// Returns a Bezier curve that results from translating the curve by the given `DVec2`.
+	pub fn translate(&self, translation: DVec2) -> Bezier {
+		self.apply_transformation(&|point| point + translation)
+	}
+
+	/// Returns a list of points where the provided `line` intersects with the Bezier curve.
+	/// - `line` - Expected to be received in the format of `[start_point, end_point]`.
```

Comment:
I think a comment like "A line passing through \`[start_point, end_point]\`." would be clearer to remove any potential for confusion with a line segment. (Or is it actually a line segment?)

Reply:
It is actually a line segment, I'll change the description to make that more clear.